lilypad/closure.py

```diff
@@ -0,0 +1,543 @@
+"""The `Closure` class."""
+
+from __future__ import annotations
+
+import ast
+import hashlib
+import importlib.metadata
+import importlib.util
+import inspect
+import json
+import os
+import site
+import subprocess
+import sys
+import tempfile
+from collections.abc import Callable
+from functools import lru_cache
+from pathlib import Path
+from textwrap import dedent
+from types import ModuleType
+from typing import Any, TypedDict, cast
+
+from packaging.requirements import Requirement
+
+
+class DependencyInfo(TypedDict):
+    version: str
+    extras: list[str] | None
+
+
+def _is_third_party(module: ModuleType, site_packages: set[str]) -> bool:
+    module_file = getattr(module, "__file__", None)
+    return (
+        module.__name__ in sys.stdlib_module_names
+        or module_file is None
+        or any(
+            str(Path(module_file).resolve()).startswith(site_pkg)
```

Comment:
I don’t have access to a Windows environment to verify, but it’s likely that this would fail to match due to differences in the case of drive letters.
In the future, should we consider adding Windows or macOS to the CI pipeline? Since this involves path manipulation, it might be necessary.

Reply:
I've added Windows to our actions to see if this still works. I'm opting to skip macOS because we run those tests on macOS locally anyway (and macOS minutes are 10x so we'll hit limits faster and likely have to pay)